libethdrivers/src/plat/imx6/imx6.c

```diff
@@ -706,14 +736,25 @@ static int init_device(imx6_eth_driver_t *dev, const nic_config_t *nic_config)
     /* Initialise the phy library */
     miiphy_init();
     /* Initialise the phy */
+#if defined(CONFIG_PLAT_SABRE) || defined(CONFIG_PLAT_WANDQ)
     phy_micrel_init();
+#elif defined(CONFIG_PLAT_NITROGEN6SX)
+    phy_atheros_init();
```

Comment:
Is this something that could eventually be queried from part of the device tree?

Reply:
Yes. In the long run, this should be dynamic. It can either be in the device tree or it could be detected dynamically when setting up the ethernet port, since a PHY is reporting the ID. Not sure what is the best thing here, because the PHY is usually no part of the SoC and it may also change over board revisions. 
Aside from this, `phy_xxx_init()` seems something form U-Boot, how PHY drivers are registered.